cloud/deployment/fromfile/fromfile.go

```diff
@@ -158,189 +148,416 @@ func CreateOrUpdate(inputFile, action string, client astro.Client, coreClient as
+			return err
 		}
-		if deploymentFromFile.Deployment.Configuration.Executor == deployment.CeleryExecutor {
-			// get defaults for min-count, max-count and concurrency from API
-			defaultOptions, err = workerqueue.GetWorkerQueueDefaultOptions(client)
-			if err != nil {
-				return astro.CreateDeploymentInput{}, astro.UpdateDeploymentInput{}, err
-			}
-			if deployment.IsDeploymentHosted(deploymentFromFile.Deployment.Configuration.DeploymentType) || deployment.IsDeploymentDedicated(deploymentFromFile.Deployment.Configuration.DeploymentType) {
-				configOptions, err = client.GetDeploymentConfig()
-				if err != nil {
-					return astro.CreateDeploymentInput{}, astro.UpdateDeploymentInput{}, err
+
+		defaultOptions = deploymentOptions.WorkerQueues
+		configOptions = deploymentOptions
+		for i := range listQueues {
+			if deployment.IsDeploymentStandard(deploymentType) || deployment.IsDeploymentDedicated(deploymentType) {
+				astroMachines := configOptions.WorkerMachines
+				for j := range astroMachines {
+					if astroMachines[j].Name == *listQueues[i].AstroMachine {
+						astroMachine = astroMachines[j]
+					}
 				}
-			}
-			for i := range listQueues {
+				var workerQueue astroplatformcore.WorkerQueueRequest
+				workerQueue.Id = &listQueues[i].Id
+				workerQueue.Name = listQueues[i].Name
+				workerQueue.IsDefault = listQueues[i].IsDefault
+				workerQueue.MaxWorkerCount = listQueues[i].MaxWorkerCount
+				workerQueue.MinWorkerCount = listQueues[i].MinWorkerCount
+				workerQueue.WorkerConcurrency = listQueues[i].WorkerConcurrency
+				workerQueue.AstroMachine = astroplatformcore.WorkerQueueRequestAstroMachine(*listQueues[i].AstroMachine)
 				// set default values if none were specified
-				a := workerqueue.SetWorkerQueueValues(listQueues[i].MinWorkerCount, listQueues[i].MaxWorkerCount, listQueues[i].WorkerConcurrency, &listQueues[i], defaultOptions)
-				if deployment.IsDeploymentHosted(deploymentFromFile.Deployment.Configuration.DeploymentType) || deployment.IsDeploymentDedicated(deploymentFromFile.Deployment.Configuration.DeploymentType) {
-					astroMachines := configOptions.AstroMachines
-					for j := range astroMachines {
-						if astroMachines[j].Type == listQueues[i].AstroMachine {
-							astroMachine = astroMachines[j]
-						}
-					}
-					// check if queue is valid
-					err = workerqueue.IsHostedCeleryWorkerQueueInputValid(a, defaultOptions, &astroMachine)
-					if err != nil {
-						return astro.CreateDeploymentInput{}, astro.UpdateDeploymentInput{}, err
-					}
-				} else {
-					// check if queue is valid
-					err = workerqueue.IsCeleryWorkerQueueInputValid(a, defaultOptions)
-					if err != nil {
-						return astro.CreateDeploymentInput{}, astro.UpdateDeploymentInput{}, err
-					}
+				a := workerqueue.SetWorkerQueueValues(listQueues[i].MinWorkerCount, listQueues[i].MaxWorkerCount, listQueues[i].WorkerConcurrency, &workerQueue, defaultOptions)
```

Comment:
can we give a real variable name instead of `a`

Reply:
changed to requestedWorkerQueues